src/odemis/driver/orsay.py

```diff
@@ -1389,3 +1440,1247 @@ def terminate(self):
+            raise ValueError("Length of Tuple VA does not match number of parameters passed.")
+        if len(self._parameters) > 1 and not self._va_is_tuple:
+            raise ValueError("Multiple parameters are passed, but VA is not of a tuple type.")
+
+        # If the VA has a range, check the Orsay server if a range of the parameter is specified and copy this range
+        if hasattr(self._va, "range"):
+            if self._va_is_tuple:
+                new_range = [list(self._va.range[0]), list(self._va.range[1])]
+            else:
+                new_range = [[self._va.range[0]], [self._va.range[1]]]
+
+            for i in range(len(self._parameters)):
+                p = self._parameters[i]
+                # Search for a lowerbound on the server
+                lowerbound = None
+                if self._minpar is not None:  # in case a minimum parameter is supplied
+                    if self._minpar[i].Actual is not None:
+                        lowerbound = self._minpar[i].Actual
+                    else:
+                        lowerbound = self._minpar[i].Target
+                if lowerbound is None:
+                    lowerbound = p.Min
+                else:
+                    if p.Min is not None and not p.Min == lowerbound:
+                        raise AssertionError("%s.Min and %s contain different, non-None values."
+                                             "Contact Orsay Physics about this!" % (p.Name, self._minpar[i].Name))
+                if lowerbound is not None:  # if a lowerbound is defined in the server
+                    new_range[0][i] = self._parameter_to_VA_value(lowerbound)  # copy it to the va
+                # Search for an upperbound on the server
+                upperbound = None
+                if self._maxpar is not None:  # in case a minimum parameter is supplied
+                    if self._maxpar[i].Actual is not None:
+                        upperbound = self._maxpar[i].Actual
+                    else:
+                        upperbound = self._maxpar[i].Target
+                if upperbound is None:
+                    upperbound = p.Max
+                else:
+                    if p.Max is not None and not p.Max == upperbound:
+                        raise AssertionError("%s.Max and %s contain different, non-None values."
+                                             "Contact Orsay Physics about this!" % (p.Name, self._maxpar[i].Name))
+                if upperbound is not None:  # if an upperbound is defined in the server
+                    new_range[1][i] = self._parameter_to_VA_value(upperbound)  # copy it to the va
+
+            if len(new_range[0]) == 1:
+                new_range = (new_range[0][0], new_range[1][0])
+            else:
+                new_range = (tuple(new_range[0]), tuple(new_range[1]))
+            # Set the range of the VA
+            self._va._value = new_range[0]
```

Comment:
Note that the Continuous VAs have a `.clip_on_range` attribute to handle it. If it's set to True (at init), then every time the range is updated, it'll automatically adjust the value, instead of complaining it's not within range. You could put this in the Connector init. Alternatively, leave this code as-is, but add a comment to explain why you put the wrong value and it's not a problem.